packages/experimental/Shimmer/src/Shimmer.tsx

```diff
@@ -0,0 +1,99 @@
+/** @jsx withSlots */
+import { useState, useEffect, useMemo } from "react";
+import { Circle, ClipPath, Defs, LinearGradient, Rect, Stop, Svg, Image } from 'react-native-svg';
+import { shimmerName, ShimmerProps, ShimmerType } from './Shimmer.types';
+import { mergeProps } from '../../../framework/merge-props/lib/mergeProps';
+import { buildUseStyling, compose, UseSlots, withSlots } from '../../../experimental/framework/lib';
+import { Animated } from 'react-native';
+import { stylingSettings } from './Shimmer.styling';
+
+export function renderAnimations(memoData: any) {
+  let startValue = useState(new Animated.Value(0))[0]
+  useEffect(() => {
+  Animated.loop(
+    Animated.sequence([
+    Animated.timing(startValue, {
+      toValue: memoData.toValue,
+      duration: memoData.duration,
+      delay: memoData.delay,
+      useNativeDriver:false
+    })]),
+  ).start()
+  })
+  return startValue;
+}
+
+const useStyling = buildUseStyling(stylingSettings);
+export const Shimmer = compose<ShimmerType>({
+  displayName: shimmerName,
+  ...stylingSettings,
+  tokensThatAreAlsoProps: ['duration', 'shimmerTintColor', 'gradientTintColor', 'toValue', 'delay'],
+  slots: {
+    root: Svg,
+    rect: Rect,
+    circle: Circle,
+  },
+  render: (props: ShimmerProps, useSlots: UseSlots<ShimmerType>) => {
+    const Root = useSlots(props).root;
+    const SvgRect = useSlots(props).rect;
+    let AnimatedLinearGradient = Animated.createAnimatedComponent(LinearGradient);
+    const tokens = useStyling(props).root;
+    const memoizedShimmerData = useMemo(
+      () => ({
+        gradientTintColor: props.gradientTintColor ? props.gradientTintColor : tokens['gradientTintColor'],
+        shimmerTintColor: props.shimmerTintColor ? props.shimmerTintColor : tokens['shimmerTintColor'],
+        containerWidth: props.containerWidth ? props.containerWidth : tokens['containerWidth'],
+        containerHeight: props.containerHeight ? props.containerHeight : tokens['containerHeight'],
+        toValue: props.toValue ? props.toValue : tokens['toValue'],
+        delay: props.delay ? props.delay : tokens['delay'],
+        duration: props.duration ? props.duration : tokens['duration'],
+      }),
+      [props.gradientTintColor, props.shimmerTintColor, props.containerWidth, props.containerHeight, props.toValue, props.delay, props.duration],
+    );
+
+    let startValue = renderAnimations(memoizedShimmerData);
+
+
+    if(props.uri != null) {
+      return (rest: ShimmerProps) => <Root {...mergeProps(props, rest)}>
+      <Defs>
+        <AnimatedLinearGradient id="gradient" x1={startValue} y1="0" x2="-1" y2="-1" >
```

Comment:
I think the more standard syntax I've seen for conditional JSX returns looks something like this

```jsx
return (
 {condition ? (
 <JSX>...</JSX>
 ) : (
 <OtherJSX>...</OtherJSX>
)}
```
It might help with lining up indentation and such.
https://reactjs.org/docs/conditional-rendering.html#inline-if-else-with-conditional-operator

Reply:
You might also be able to do that inline and conditionally render just the <Image> or <Rect> based on props.uri. Not sure if the JSX rules works with SVG elements though 🤔

Reply:
I removed the conditional check and added some more slots